include/envoy/server/drain_manager.h

```diff
@@ -3,17 +3,35 @@
 #include <functional>
 #include <memory>
 
+#include "envoy/config/listener/v3/listener.pb.h"
+#include "envoy/event/dispatcher.h"
 #include "envoy/network/drain_decision.h"
 
 namespace Envoy {
 namespace Server {
 
+class DrainManager;
+using DrainManagerPtr = std::unique_ptr<DrainManager>;
+using DrainManagerSharedPtr = std::shared_ptr<DrainManager>;
+
 /**
  * Handles connection draining. This concept is used globally during hot restart / server draining
  * as well as on individual listeners when they are being dynamically removed.
  */
 class DrainManager : public Network::DrainDecision {
 public:
+  /**
+   * @brief Create a child drain-manager. Will proxy the drain status from the parent, but can also
+   * be used to enact local draining.
+   *
+   * @param dispatcher Dispatcher for the current thread in which the new child drain-manager will
+   * exist.
```

Comment:
Would be useful to expand this comment about intended use of child managers and how their behavior differs from their parent, e.i. ignores the drain time argument

Reply:
Comment expanded.